zpm/publisher_abs.go

```diff
@@ -485,6 +545,14 @@ func (a *ABSPublisher) publish(osarch *zps.OsArch, pkgFiles []string, zpkgs []*z
 			return err
 		}
 
+		medatabaDbData := make([]byte, 0)
+		_, err = metadataUp.Read(medatabaDbData)
+		if err != nil {
+			return fmt.Errorf("unable to read downloaded file: %s", a.uri.Path)
+		}
+
+		eTag = md5.Sum(medatabaDbData)
```

Comment:
Somewhere below, after the upload succeeds the new eTag should be updated by calling `locker.UnlockWithEtag`

This applies to the other publishers as well.

Reply:
it's called in defer now

```
defer locker.UnlockWithEtag(&eTag)
```

I do believe it should be fine since it's a pointer